nuxeo-retention/src/test/java/org/nuxeo/retention/test/DownloadAssetAction.java

```diff
@@ -0,0 +1,109 @@
+
+/**
+ * @since 11.5
+ */
+public class DownloadAssetAction implements StreamProcessorTopology {
+
+    public static final String ACTION_NAME = "downloadAssetAction";
+
+    public static final String ACTION_FULL_NAME = "frameio/" + ACTION_NAME;
+
+    public static final String PARAM_DESC = "description";
+
+    @Override
+    public Topology getTopology(Map<String, String> options) {
+        return Topology.builder()
+                       .addComputation(DownloadAssetComputation::new,
+                               List.of(INPUT_1 + ":" + ACTION_FULL_NAME, OUTPUT_1 + ":" + STATUS_STREAM))
+                       .build();
+    }
+
+    public static class DownloadAssetComputation extends AbstractBulkComputation {
+
+        protected String description;
+
+        public DownloadAssetComputation() {
+            super(ACTION_FULL_NAME);
+        }
+
+        @Override
+        public void startBucket(String bucketKey) {
+            BulkCommand command = getCurrentCommand();
+            description = command.getParam(PARAM_DESC);
+        }
+
+        @Override
+        protected void compute(CoreSession session, List<String> ids, Map<String, Serializable> properties) {
+            // Computation should be peer an asset download.
+            if (ids.size() > 1) {
+                throw new IllegalArgumentException("Many folders destinations are given for the asset creation !");
+            }
+
+            HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+            String url = (String) properties.get("url");
+            String fileName = FilenameUtils.getBaseName(url);
+            String extension = FilenameUtils.getExtension(url);
+            HttpGet request = new HttpGet(url);
+            Path destination = null;
+            try (CloseableHttpClient httpClient = httpClientBuilder.build();
+                    CloseableHttpResponse response = httpClient.execute(request);
+                    InputStream is = response.getEntity().getContent()) {
+                // TODO Florent/Nelson?
+                // Here for testing purpose I am using the temporary file as a destination. Should we consider a
+                // specific directory destination by configuration
+                destination = Files.createTempFile(fileName, "." + extension);
+                // Should we consider the usage of ReadableByteChannel and FileChannel.transferFrom
+                Files.copy(is, destination, StandardCopyOption.REPLACE_EXISTING);
+            } catch (IOException e) {
+                throw new NuxeoException(e);
+            }
+
+            // Use the File Manager Import operation to create the blob's document
+            try (OperationContext ctx = new OperationContext(session)) {
+                // TODO Florent/Nelson should we consider putting the temp blob in the transient store for GC purposes
+                Blob blob = new FileBlob(destination.toFile(), true);
```

Comment:
Instead of `new FileBlob` use `Blobs.createBlobWithExtension(extension)` earlier to create the underlying tmp file, then use `blob.getFile()` to get the temporary file that was just created and use it to be filled from the url.
This will also deal with the cleanup as the underlying `FileBlob` created will be flagged as temporary, and tracked for deletion if nobody uses it anymore (and also it optimizes copies).